v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/PubSubToElasticsearch.java

```diff
@@ -142,24 +148,27 @@ public static PipelineResult run(PubSubToElasticsearchOptions options) {
     /*
      * Step #3a: Write Json documents into Elasticsearch using {@link ElasticsearchTransforms.WriteToElasticsearch}.
      */
-    convertedPubsubMessages
-        .get(TRANSFORM_OUT)
-        .apply(
-            "GetJsonDocuments",
-            MapElements.into(TypeDescriptors.strings()).via(FailsafeElement::getPayload))
-        .apply("Insert metadata", new ProcessEventMetadata())
-        .apply(
-            "WriteToElasticsearch",
-            WriteToElasticsearch.newBuilder()
-                .setOptions(options.as(PubSubToElasticsearchOptions.class))
-                .build());
+    PCollection<String> failedMessages =
+        convertedPubsubMessages
+            .get(TRANSFORM_OUT)
+            .apply(
+                "GetJsonDocuments",
+                MapElements.into(TypeDescriptors.strings()).via(FailsafeElement::getPayload))
+            .apply("Validate JSON fields", new ProcessValidateJsonFields())
+            .apply("Insert metadata", new ProcessEventMetadata())
+            .apply(
+                "WriteToElasticsearch",
+                WriteToElasticsearch.newBuilder()
+                    .setOptions(options.as(PubSubToElasticsearchOptions.class))
+                    .build());
 
     /*
      * Step 3b: Write elements that failed processing to error output PubSub topic via {@link PubSubIO}.
      */
-    convertedPubsubMessages
```

Comment:
Are we ignoring the errors (TRANSFORM_ERROROUTPUT_OUT ) returned by ConvertMessageToJsonDocument transform step?